integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java

```diff
@@ -332,4 +330,43 @@ public void testKotlinJvmWithSubmoduleProject() throws Exception {
     assertThat(binaries).containsExactly(baseDir.resolve("submodule/build/classes/kotlin/main").toString());
     assertThat(sources).containsExactly(baseDir.resolve("submodule/src/main/kotlin").toString());
   }
+
+  @Test
+  public void testScanAllOnMultiModuleWithSubModulesProject() throws Exception {
+    Properties props = runGradlewSonarSimulationModeWithEnv("/multi-module-with-submodules", emptyMap(), "compileJava", "compileTestJava");
+
+    Path baseDir = Paths.get(props.getProperty("sonar.projectBaseDir"));
+    assertThat(baseDir.getFileName().toString()).hasToString("multi-module-with-submodules");
+
+    String scanAll = props.getProperty("sonar.gradle.scanAll");
+    assertThat(scanAll).isEqualTo("true");
+
+    // checking that the skippedModules are not included in the sonar.modules property, while the other non-skipped nested modules are
+    assertThat(props.getProperty("sonar.modules")).isEqualTo(":module");
+    assertThat(props.getProperty(":module.sonar.modules")).isEqualTo(":module:submodule");
+
+    // checking that the expected sources are collected when the scanAll property is set to true
+    String[] sources = props.getProperty("sonar.sources").split(",");
+    assertThat(sources)
+      .containsExactlyInAnyOrder(
+        baseDir.resolve("build.gradle.kts").toString(),
+        baseDir.resolve("settings.gradle.kts").toString(),
+        baseDir.resolve("module/build.gradle.kts").toString(),
+        baseDir.resolve("module/submodule/build.gradle.kts").toString(),
+        baseDir.resolve("module/submodule/submoduleScript.sh").toString(),
+        baseDir.resolve("gradlew").toString(),
+        baseDir.resolve("gradlew.bat").toString(),
+        // Will be fixed by SCANGRADLE-149
+        baseDir.resolve("skippedModule/skippedSubmodule/build.gradle.kts").toString(),
+        baseDir.resolve("gradle/wrapper/gradle-wrapper.properties").toString())
+      .doesNotContain(
```

Comment:
Let's move the closing parenthesis closer to the next line to make it clearer that  `.doesNotContain` starts a brand new assertion
```suggestion
        baseDir.resolve("gradle/wrapper/gradle-wrapper.properties").toString()
      ).doesNotContain(
```